File: test/fixtures/go-app-shipping/shipping_test.go
package shipping

import "testing"

func TestOrderUnderThresholdPaysForShipping(t *testing.T) {
	if IsFreeShipping(4999) {
		t.Errorf("IsFreeShipping(4999) = true, want false")
	}
	if got := ShippingCost(4999); got != 499 {
		t.Errorf("ShippingCost(4999) = %d, want 499", got)
	}
}

func TestOrderOverThresholdShipsFree(t *testing.T) {
	if !IsFreeShipping(5001) {
		t.Errorf("IsFreeShipping(5001) = false, want true")
	}
}

func TestOrderExactlyAtThresholdShipsFree(t *testing.T) {
	// $50.00 is the advertised cutoff, not $50.01. This is the seeded bug.
	if !IsFreeShipping(5000) {
		t.Errorf("IsFreeShipping(5000) = false, want true")
	}
	if got := ShippingCost(5000); got != 0 {
		t.Errorf("ShippingCost(5000) = %d, want 0", got)
	}
}

File: test/fixtures/ts-app-shipping/shipping.test.js
import assert from 'node:assert/strict';
import { test } from 'node:test';
import { isFreeShipping, shippingCost } from './shipping.js';

test('an order under the threshold pays for shipping', () => {
  assert.equal(isFreeShipping(4999), false);
  assert.equal(shippingCost(4999), 499);
});

test('an order over the threshold ships free', () => {
  assert.equal(isFreeShipping(5001), true);
});

test('an order exactly at the threshold ships free', () => {
  // $50.00 is the advertised cutoff, not $50.01. This is the seeded bug.
  assert.equal(isFreeShipping(5000), true);
  assert.equal(shippingCost(5000), 0);
});
